src/content.js

```diff
@@ -49,23 +49,18 @@ function analyzeImage(image) {
   chrome.runtime.sendMessage({ url: image.src, url2: image.src }, response => {
     console.log(`Prediction result is ${response ? response.result : 'undefined'} for image ${image.src}`);
     if (response && response.result === false) {
-      image.style.visibility = 'visible'
+      image.style.visibility = 'visible';
     } else {
       image.__isNSFW = true
     }
   });
 }
 
-// Call function when DOM is loaded
-window.addEventListener("load", () => {
-  clasifyImages()
-
-  // @todo hanlde with javascript render delay
-  const timesToJSrun = [500, 1000, 2000, 3000]
-  for (let i = 0; i < timesToJSrun.length; i++) {
-    setTimeout(() => { clasifyImages() }, timesToJSrun[i]);
-  }
+//Call function when changes made in DOM tree
```

Comment:
When you load a web page at first the images are hidden without any lag. But when the same page is reloaded, the filter does not work. Also I am experiencing considerable slowing down of my browser while running the extension that did not happen before

Reply:
Which page are you testing with? And when you say reload, does that mean refreshing?(I tried https://scrolller.com/nsfw) and it seems to work. Is there a way to see the slowing down of the browser? I didn't notice it.

Reply:
I tested with Google Images. I tried to reload/refresh on the browser and the filter did not seem to work then. I tried with the current MASTER branch and the PR and there was a considerable lag between them. Could you test it out thoroughly?